Global demo_mode flag

gasse opened the pull request:
- [x] `browsergym.core.action.set_global_demo_mode(demo_mode: bool)`
 - [x] `keyboard_type()` action demo_mode fix
 - [x] `fill()` action demo_mode fix
 - [x] ci tests

recursix commented on browsergym/core/src/browsergym/core/action/highlevel.py:
```diff
@@ -101,7 +97,9 @@ class HighLevelAction:
 
+    # static class variables
     ActionSubset = Literal["chat", "infeas", "bid", "coord", "nav", "tab", "custom"]
+    default_demo_mode: Literal["off", "default", "all_blue", "only_visible_elements"] = "off"
```
shouldn't we put it at the module level in case someone defines a new ActionSet that calls the same sub-functions?
or you think they would always subclass HighLevelActionSet?

gasse replied:
I though about it, but I think it's more explicit and less confusing that way

### Option 1

```python
browsergym.core.action.highlevel.HighLevelActionSet.default_demo_mode = True
```
-> this will set the default demo_mode of this action set only

### Option 2

```python
browsergym.core.action.default_demo_mode = True
```
-> this will set the demo_mode of any action set (NOT TRUE)


Maybe more importantly, the demo_mode choices are very arbitrary and specific to HighLevelActionSet
```python
"off", "default", "all_blue", "only_visible_elements"
```

gasse replied:
Ok I think I have an idea that will make both of us happy.

gasse commented:
@recursix I changed to a global flag, can you make a new pass on the code?
```python
browsergym.core.action.set_global_demo_mode(global_demo_mode: bool)
```

I also expanded the demo_mode tests, which exposed an existing bug with the fill() action, which I fixed with a minor refactor.

recursix reviewed (approved):
mostly blind approved
